### PR TITLE
Add pagination to get_workflow_history

### PR DESCRIPTION
**Pull request by cduanfigma**
Uses pagination in `get_workflow_history` so that it fetches the complete history result. 

Review without whitespace. 

Fixes https://github.com/coinbase/temporal-ruby/issues/289

### CODE REVIEW

File: lib/temporal/client.rb [hunk 1]
```diff
@@ -398,13 +398,23 @@ def fail_activity(async_token, exception)
...
+            next_page_token: next_page_token,
+          )
+        events.concat(response.history.events.to_a)
+        next_page_token = response.next_page_token
+
+        break if next_page_token.empty?
+      end
 
-      Workflow::History.new(history_response.history.events)
+      Workflow::History.new(response.history.events)
```
Comment (DeRauk) [inline lib/temporal/client.rb:417]: Did you mean to create the workflow history object from the `events` property? Creating it from `response.history.events` is incorrect.

File: spec/unit/lib/temporal/client_spec.rb [hunk 2]
```diff
@@ -895,6 +895,32 @@ class NamespacedWorkflow < Temporal::Workflow
...
+        .ordered
+
+      expect(connection)
+        .to have_received(:get_workflow_execution_history)
+        .with(namespace: namespace, workflow_id: workflow_id, run_id: run_id, next_page_token: 'a')
+        .ordered
+
+      expect(connection).to have_received(:get_workflow_execution_history).exactly(2).times
+    end
+  end
```
Comment (DeRauk) [inline spec/unit/lib/temporal/client_spec.rb:922]: This test is failing in CI, could you take a look?

Comment (cduanfigma) [reply spec/unit/lib/temporal/client_spec.rb:922]: should be fixed!

**Comment by efroese-figma** [comment]
Hey! Just checking in on this PR. @DeRauk,  @cduanfigma addressed the comments you left. Is there anything else we should do to get this merged?

**Comment by DeRauk** [comment]
@cj-cb could you take a look at this one?

**Comment by cj-cb** [comment]
@cduanfigma Thank you for the PR.  Can you please resolve the conflicts?

**Comment by cduanfigma** [comment]
@cj-cb conflicts are resolved!

**Review by cj-cb — APPROVED** [review]

**Comment by cj-cb** [comment]
@cduanfigma Looks like tests are failing.  Do you mind pulling in the latest in master again to get a fix?

**Comment by cduanfigma** [comment]
@cj-cb done, this should pick up the changes in https://github.com/coinbase/temporal-ruby/pull/308

**Pull request merged**